### PR TITLE
Bug 1807430: don't sync on invalid configuration

### PR DESCRIPTION
**Pull request by simonpasquier**
cc @openshift/openshift-team-monitoring

### CODE REVIEW

**Comment by simonpasquier** [comment]
@lilic yes, when an invalid configuration is provided, the operator will stop its synchronization job instead of using the default (empty) configuration. The operator will be marked as degraded so the admin is notified and able to fix the issue.

File: test/e2e/config_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
```
Comment (lilic) [inline test/e2e/config_test.go:1]: 2020 sadly :)

File: test/e2e/config_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/cluster-monitoring-operator/test/e2e/framework"
+)
+
+func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
+	// Enable user workload monitoring to assess that an invalid configuration
```
Comment (lilic) [inline test/e2e/config_test.go:30]: Do we need to enable this just to test configuration, we use same configmap to configure both workloads, or is it some specific edge case we want to cover?

Comment (simonpasquier) [reply test/e2e/config_test.go:30]: The goal here is to verify that a valid "custom" configuration isn't rolled back after we push an invalid config. I could have used another setting (like volume claim, resource requests, ...) but enabling user workload monitoring and checking that the `prometheus-user-workload` statefulset still exists was the "simplest" thing to do from my point of view.

File: test/e2e/config_test.go (hunk 1 above)
Comment (simonpasquier) [reply test/e2e/config_test.go]: :eyeglasses:

File: test/e2e/config_test.go (hunk 2 above)
Comment (lilic) [reply test/e2e/config_test.go:30]: Okay makes sense, thanks for explaining!
